### ISSUE TITLE
Logic bug in courseActions.js - CREATE_COURSE_SUCCESS isn't called in demo

### ISSUE THREAD

**Issue by coryhouse** [coryhouse/pluralsight-redux-app-used-to-build-script#4]
There's a logic issue in saveCourse in courseActions.js. I'm overwriting the value of course passed in with the result from the mock API, so the check for course.id will always return true. This leads to updateCourseSuccess always being called.

To fix, change saveCourse in courseActions.js to this:

```
export function saveCourse(course) {
  return function (dispatch, getState) {
    dispatch(beginAjaxCall());
    debugger;
    return CourseApi.saveCourse(course).then(savedCourse => {
      course.id ? dispatch(updateCourseSuccess(savedCourse)) : dispatch(createCourseSuccess(savedCourse));
    }).catch(error => {
      dispatch(ajaxCallError(error));
      throw(error);
    });
  };
}
```

Note how I'm naming the response from saveCourse `savedCourse` to avoid overwriting the course passed in. This way, the check for courseId in the response handler can accurately determine if it's working with a create or an update.

**Comment by coryhouse** [comment]
Note: This is fixed in #1

**Comment by coryhouse** [comment]
Closing since 3/12/19 course update addressed this issue.
